### PR TITLE
CE-606 Bug fix for validation

### PR DESCRIPTION
**Pull request by cnesmithsalus**
# Description

Please provide a summary of the change and the issue fixed. Please include relevant context. List dependency changes.

Fixes # (issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Further comments

---

Thanks for the PR!

Any successful deployments (not always required) will be available below.
[Backend](https://nr-compliance-enforcement-338-backend.apps.silver.devops.gov.bc.ca/) available
[Frontend](https://nr-compliance-enforcement-338-frontend.apps.silver.devops.gov.bc.ca/) available

Once merged, code will be promoted and handed off to following workflow run.
[Main Merge Workflow](https://github.com/bcgov/nr-compliance-enforcement/actions/workflows/merge-main.yml)

### CODE REVIEW

File: frontend/src/app/components/containers/complaints/outcomes/animal-outcomes/drug-authorization.tsx [hunk 1]
```diff
@@ -35,12 +35,12 @@ export const DrugAuthorization: FC<Props> = ({ agency, drugAuthorization, update
   const handleAuthorizedByChange = (input: string | undefined) => {
     setAuthorizedBy(input);
     const newDrugAuth: DrugAuthorizationType = { officer: input ?? "", date: authorizedOn ?? undefined };
-    update(newDrugAuth);
+    update(newDrugAuth, "officer");
   };
 
   const handleAuthorizedOnChange = (input: Date | undefined | null) => {
```
Comment (marqueone-ps) [inline frontend/src/app/components/containers/complaints/outcomes/animal-outcomes/drug-authorization.tsx:41]: is there a need for the null type?

File: frontend/src/app/components/containers/complaints/outcomes/animal-outcomes/edit-animal-outcome.tsx [hunk 2]
```diff
@@ -374,6 +374,23 @@ export const EditAnimalOutcome: FC<EditAnimalOutcomeProps> = ({
     return isValid;
   };
 
+  const updateDrugAuthorizationFromInput = (drugAuthorization: DrugAuthorization | undefined, type: string) => {
```
Comment (marqueone-ps) [inline frontend/src/app/components/containers/complaints/outcomes/animal-outcomes/edit-animal-outcome.tsx:377]: This function has me scratching my head, it looks very very similar to `updateDrugAuthorization` plus I can't seem to grok where the `isValid` is coming from.

File: frontend/src/app/components/containers/complaints/outcomes/animal-outcomes/drug-authorization.tsx (hunk 1 above)
Comment (cnesmithsalus) [reply frontend/src/app/components/containers/complaints/outcomes/animal-outcomes/drug-authorization.tsx:41]: If I take the null away it behaves kind of weird when you blank out the date

File: frontend/src/app/components/containers/complaints/outcomes/animal-outcomes/edit-animal-outcome.tsx (hunk 2 above)
Comment (cnesmithsalus) [reply frontend/src/app/components/containers/complaints/outcomes/animal-outcomes/edit-animal-outcome.tsx]: cleaned this up a bit to reduce duplication

**Review by marqueone-ps — APPROVED** [review]

**Review by afwilcox — APPROVED** [review]

**Pull request merged**